sigh/lib/assets/resign.sh

```diff
@@ -787,7 +787,7 @@ function resign {
         OLD_BUNDLE_ID="$(PlistBuddy -c "Print :CFBundleIdentifier" "$TEMP_DIR/oldInfo.plist")"
         NEW_BUNDLE_ID="$(bundle_id_for_provison "$NEW_PROVISION")"
         log "Replacing old bundle ID '$OLD_BUNDLE_ID' with new bundle ID '$NEW_BUNDLE_ID' in patched entitlements"
-        sed -i .bak "s/$OLD_BUNDLE_ID/$NEW_BUNDLE_ID/g" "$PATCHED_ENTITLEMENTS"
+        sed -i .bak "s!${OLD_BUNDLE_ID}</string>!${NEW_BUNDLE_ID}</string>!g" "$PATCHED_ENTITLEMENTS"
```

Comment:
Yes, in theory (and I considered that), but it's problematic because $OLD_BUNDLE_ID and $NEW_BUNDLE_ID don't include the full team id prefix, so the matching wouldn't work. i.e.

Entitlements file looks like:

<string>AB1GP98Q19.com.example.foo</string>

But OLD_BUNDLE_ID and NEW_BUNDLE_ID are just the com.example.foo.

So this would be a bigger change that I'm not confident to make :-).

Reply:
Ok, good point! Maybe just add a quick code comment here (how does one add comments to a shell script?) as a recognized possible bug cause - as this is so, so unprobable to happen I think we are good not handling it.

Reply:
Thanks @janpio - good idea and comment added.

I actually wondered if this code ever actually does anything useful anyway, because I think the bundle ID should always be correct in the entitlements file (because it comes from the new provisioning profile). There may be another case I've missed though (possibly related to the case where use app entitlements isn't enabled), so I'm not proposing we remove it.

Reply:
[sorry, pushed once more to fix a typo - not doing well today!]

Reply:
That I can't say - I don't _really_ understand how this script works at all. So I am good with fixing this possible bug (and #9550 indicates it used to be a problem in special cases).

Reply:
Indeed - the case I was seeing where have an appstore build: `com.example.foo` and a resigned enterprise version `com.example.fooEnterprise`. Thanks for reviewing!